src/pacman/query.c

```diff
@@ -70,6 +73,32 @@ static int search_path(char **filename, struct stat *bufptr)
 
 		if(lstat(fullname, bufptr) == 0) {
 			free(*filename);
+
+			/* We check if the file actually has an .exe suffix by lstat'ing
+			 * the same path with .exe suffix and checking if the inode
+			 * numbers are the same: If they are, we return the name with
+			 * .exe appended, because otherwise the package ownership check
+			 * breaks/fails.
+			 */
+			strcat(fullname, ".exe");
+			struct stat sbExe;
+			if (lstat(fullname, &sbExe) == 0) {
+				if (sbExe.st_ino == bufptr->st_ino) {
```

Comment:
nitpick, st_dev needs to also be the same

Reply:
Fixed!